extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/clockprovider/HibernateValidatorClockProvider.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.hibernate.validator.runtime.clockprovider;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+
+import jakarta.validation.ClockProvider;
+
+/**
+ * A Quarkus-specific clock provider that can provide a clock based on a runtime system time zone.
+ */
+public class HibernateValidatorClockProvider implements ClockProvider {
+
+    public static final HibernateValidatorClockProvider INSTANCE = new HibernateValidatorClockProvider();
+
+    private static final HibernateClock clock = new HibernateClock();
+
+    private HibernateValidatorClockProvider() {
+    }
+
+    @Override
+    public Clock getClock() {
+        return clock;
+    }
+
+    private static class HibernateClock extends Clock {
```

Comment:
Don't you think the constant could live in this very class and we could avoid having one additional class?

Also, maybe we need a better name `RuntimeReinitializedDefaultClock`?

Reply:
You mean the timezone constant right?
I've started with that but the native build failed in that case. I don't have the exact message, but essentially it was saying that the clock provider / clock instances were created and such class cannot be reinitialised (at least that was my understanding of the failure 🙈)
I'll update the clock class name and add the error I've seen once I'm around the workstation 

Thanks for taking a look 😊

Reply:
Yeah OK, no need to dig if it wasn't working. Maybe add a comment in the code so that our futur selves don't try to simplify it later.

Reply:
An error (if we try to reinit the clock provider itself) for the history 😃:
```
Error: An object of type 'io.quarkus.hibernate.validator.runtime.clockprovider.HibernateValidatorClockProvider' was found in the image heap. This type, however, is marked for initialization at image run time for the following reason: classes are initialized at run time by default.
This is not allowed for correctness reasons: All objects that are stored in the image heap must be initialized at build time.

You now have two options to resolve this:

1) If it is intended that objects of type 'io.quarkus.hibernate.validator.runtime.clockprovider.HibernateValidatorClockProvider' are persisted in the image heap, add 

    '--initialize-at-build-time=io.quarkus.hibernate.validator.runtime.clockprovider.HibernateValidatorClockProvider'

to the native-image arguments. Note that initializing new types can store additional objects to the heap. It is advised to check the static fields of 'io.quarkus.hibernate.validator.runtime.clockprovider.HibernateValidatorClockProvider' to see if they are safe for build-time initialization,  and that they do not contain any sensitive data that should not become part of the image.

2) If these objects should not be stored in the image heap, you can use 

    '--trace-object-instantiation=io.quarkus.hibernate.validator.runtime.clockprovider.HibernateValidatorClockProvider'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with 

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.

If you are seeing this message after enabling '--strict-image-heap', this means that some objects ended up in the image heap without their type being marked with --initialize-at-build-time.
To fix this, include '--initialize-at-build-time=io.quarkus.hibernate.validator.runtime.clockprovider.HibernateValidatorClockProvider' in your configuration. If the classes do not originate from your code, it is advised to update all library or framework dependencies to the latest version before addressing this error.
Please address this problem to be prepared for future releases of GraalVM.

The following detailed trace displays from which field in the code the object was reached.
Trace: Object was reached by
  reading field org.hibernate.validator.internal.engine.constraintvalidation.HibernateConstraintValidatorInitializationContextImpl.clockProvider of constant 
    org.hibernate.validator.internal.engine.constraintvalidation.HibernateConstraintValidatorInitializationContextImpl@127ff225: org.hibernate.validator.internal.engine.constraintvalidation.HibernateConstraint...
  reading field org.hibernate.validator.internal.engine.ValidatorFactoryScopedContext.constraintValidatorInitializationContext of constant 
    org.hibernate.validator.internal.engine.ValidatorFactoryScopedContext@41d08dc7: org.hibernate.validator.internal.engine.ValidatorFactoryScopedContext@41d08dc7
  reading field org.hibernate.validator.internal.engine.PredefinedScopeValidatorFactoryImpl.validatorFactoryScopedContext of constant 
    org.hibernate.validator.internal.engine.PredefinedScopeValidatorFactoryImpl@3743cd98: org.hibernate.validator.internal.engine.PredefinedScopeValidatorFactoryImpl@3743...
  reading field io.quarkus.hibernate.validator.runtime.HibernateValidatorRecorder$2$1.val$validatorFactory of constant 
    io.quarkus.hibernate.validator.runtime.HibernateValidatorRecorder$2$1@4a89722e: io.quarkus.hibernate.validator.runtime.HibernateValidatorRecorder$2$1@4a89722e
  reading field java.util.concurrent.ConcurrentLinkedDeque$Node.item of constant 
    java.util.concurrent.ConcurrentLinkedDeque$Node@368bcdb6: java.util.concurrent.ConcurrentLinkedDeque$Node@368bcdb6
  reading field java.util.concurrent.ConcurrentLinkedDeque$Node.prev of constant 
    java.util.concurrent.ConcurrentLinkedDeque$Node@38f22cb4: java.util.concurrent.ConcurrentLinkedDeque$Node@38f22cb4
  reading field java.util.concurrent.ConcurrentLinkedDeque.head of constant 
    java.util.concurrent.ConcurrentLinkedDeque@55bfd898: [io.quarkus.hibernate.validator.runtime.HibernateValidatorRecorder$2$1@4a89722e,...
  reading field io.quarkus.runtime.StartupContext.shutdownTasks of constant 
    io.quarkus.runtime.StartupContext@5b8adfe6: io.quarkus.runtime.StartupContext@5b8adfe6
  reading static field io.quarkus.runner.ApplicationImpl.STARTUP_CONTEXT
    at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
  parsing method io.quarkus.runner.ApplicationImpl.doStop(Unknown Source) reachable via the parsing context
    at io.quarkus.runtime.Application.stop(Application.java:208)
    at io.quarkus.runtime.Application.close(Application.java:137)
    at java.io.FileDescriptor.closeAll(FileDescriptor.java:352)
    at java.io.FileInputStream.close(FileInputStream.java:524)
    at jdk.xml.internal.SecuritySupport.loadProperties(SecuritySupport.java:239)
    at jdk.xml.internal.SecuritySupport.readConfig(SecuritySupport.java:217)
    at com.sun.org.apache.xerces.internal.utils.XMLSecurityPropertyManager.getSystemProperty(XMLSecurityPropertyManager.java:234)
    at com.sun.org.apache.xerces.internal.utils.XMLSecurityPropertyManager.readSystemProperties(XMLSecurityPropertyManager.java:215)
    at com.sun.org.apache.xerces.internal.utils.XMLSecurityPropertyManager.<init>(XMLSecurityPropertyManager.java:96)
    at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl.<init>(DocumentBuilderFactoryImpl.java:58)
    at com.oracle.svm.core.code.FactoryMethodHolder.DocumentBuilderFactoryImpl_constructor_fe18771c43c781794fd627c47aabb2c91c25ceff(generated:0)
    at static root method.(Unknown Source)
```